### PR TITLE
feat: use package attributes to determine lint preset

### LINKED ISSUE #56: [web-scripts] only include JSX a11y if React is installed in consumer
https://github.com/kentcdodds/kcd-scripts/blob/b1ee9f87278e0cdaf5a444556d3e4d5ce1342bd3/src/config/eslintrc.js#L7-L8

### PR DESCRIPTION
**Pull request by fastfrwrd**
By checking the package that called web-scripts or eslint-config, we can determine whether we should check for React or TypeScript stuff or not.

fix #56

### CODE REVIEW

File: packages/eslint-config/index.js [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
+  { type: 'dependency', dependency: 'typescript', dependencyType: 'dev' },
+  { type: 'file', pattern: 'tsconfig.json' },
+]);
+
 module.exports = {
   extends: [
     '@spotify/eslint-config-base',
-    '@spotify/eslint-config-react',
-    '@spotify/eslint-config-typescript',
+    hasReact ? '@spotify/eslint-config-react' : null,
```
Comment (fastfrwrd) [inline packages/eslint-config/index.js]: based on the setup from `kcd-scripts`: https://github.com/kentcdodds/kcd-scripts/blob/b1ee9f87278e0cdaf5a444556d3e4d5ce1342bd3/src/config/eslintrc.js#L7

File: packages/web-scripts-utils/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "@spotify/web-scripts-utils",
+  "version": "2.0.0",
+  "private": true,
```
Comment (fastfrwrd) [inline packages/web-scripts-utils/package.json]: private dependency means it should not be published when everything else is published, it just is packaged in at release (hopefully).

File: packages/eslint-config/index.js [hunk 3]
```diff
@@ -10,6 +10,16 @@ const hasTypescript = hasConfig([
...
 
+let config;
+
+if (hasReact) {
+  config = {
+    react: {
+      version: 'detect',
+    },
+  };
+}
```
Comment (fastfrwrd) [inline packages/eslint-config/index.js]: @jomaxx this should fix the thing where you must specific a react version.

Comment (jomaxx) [reply packages/eslint-config/index.js]: awesome! thanks for doing this

**Pull request merged**